hbase-server/src/main/java/org/apache/hadoop/hbase/master/ServerManager.java

```diff
@@ -227,10 +227,6 @@ ServerName regionServerStartup(RegionServerStartupRequest request, int versionNu
     ServerName sn = ServerName.valueOf(hostname, request.getPort(), request.getServerStartCode());
     checkClockSkew(sn, request.getServerCurrentTime());
     checkIsDead(sn, "STARTUP");
-    if (!checkAndRecordNewServer(sn, ServerMetricsBuilder.of(sn, versionNumber, version))) {
```

Comment:
method javadoc needs updation. Also please add the detail that the regionserver will be added in it's first report (for readers).